internal/kafka/internal/handlers/cluster.go

```diff
@@ -132,3 +134,38 @@ func (h clusterHandler) List(w http.ResponseWriter, r *http.Request) {
 
 	handlers.HandleList(w, r, cfg)
 }
+
+func (h clusterHandler) DeregisterEnterpriseCluster(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	clusterID := mux.Vars(r)["id"]
+	forceDeleteFlag := r.URL.Query().Get("force")
+
+	if forceDeleteFlag == "" {
+		forceDeleteFlag = "false"
+	}
+
+	forceBoolValue, err := strconv.ParseBool(forceDeleteFlag)
+
+	validateForceParam := func() handlers.Validate {
+		return func() *errors.ServiceError {
+			if err != nil {
+				return errors.GeneralError("failed to parse query param force: %s", forceDeleteFlag)
+			}
+			return nil
+		}
+	}
+
+	cfg := &handlers.HandlerConfig{
+		Validate: []handlers.Validate{
+			handlers.ValidateAsyncEnabled(r, "deleting enterprise cluster"),
+			validateForceParam(),
+			ValidateKafkaClaims(ctx, ValidateOrganisationId()),
+			validateEnterpriseClusterEligibleForDeregistration(ctx, clusterID, forceBoolValue, h.clusterService),
```

Comment:
The admin has to be of the same organization as the one who registered the cluster i.e we also should have a check for org id `context.orgID === cluster.organizationID`

Reply:
I was thinking about it, but currently we don't have it for registration endpoint, so if we add it here, we should also add it in the other endpoint

Reply:
+1 on adding it everywhere. 

For the registration endpoint I seem to remember there was a JIRA for this. It could be done as part of this PR or a follow up one but for this PR it is essential that we add those two checks.

Reply:
@machi1990 - I have addressed orgAdmin comment. orgID match is already implemented AFAIK

Reply:
@pawelpaszki the is org admin check is not there. I think you added it only for the registration endpoint.

> orgID match is already implemented AFAIK

I do not see this check. Can you point me to where it is done? Thanks

Reply:
in the validation: `internal/kafka/internal/handlers/validation.go#L435` you check ID match and L425 checks for org admin

Reply:
Thanks. I see all the checks now, earlier on I was on an outdated patch.